### PR TITLE
[evm] Upgrade to Shanghai at SumatraHeight

### PR DESCRIPTION
**Pull request by envestcc**
# Description

Upgrade evm from Bellatrix to Shanghai at SumatraHeight. The main changes include: 
- Updating go-ethereum. 
- Activating `Merge` and `Shanghai` at Sumatra height. 
- Disabling EIP-4399 because there is not suitable random number implementation yet. 
- Deprecating `PrepareAccessList` and using `Prepare` instead. For background, refer to #3964.

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [x] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by CoderZhi** [comment]
split it into multiple prs

File: action/protocol/execution/evm/evm_test.go [hunk 1]
```diff
@@ -257,26 +265,32 @@ func TestConstantinople(t *testing.T) {
 		)
 		require.NoError(err)
 
+		now := time.Now()
+		getBlockTime := func(height uint64) (time.Time, error) {
+			return now.Add(time.Duration(height) * time.Second * 5), nil
+		}
```
Comment (dustinxie) [inline action/protocol/execution/evm/evm_test.go:271]: these 2 lines can move to in front of `if`

File: action/protocol/execution/evm/evm.go [hunk 2]
```diff
@@ -176,14 +176,23 @@ func newParams(
...
-		Time:        new(big.Int).SetInt64(blkCtx.BlockTimeStamp.Unix()),
+		Time:        new(big.Int).SetInt64(blkCtx.BlockTimeStamp.Unix()).Uint64(),
 		Difficulty:  new(big.Int).SetUint64(uint64(50)),
 		BaseFee:     new(big.Int),
 	}
+	if g.IsSumatra(blkCtx.BlockHeight) {
+		// Random opcode (EIP-4399) is not supported on IoTeX
+		zeroHash := common.BytesToHash(nil)
+		context.Random = &zeroHash
+	}
```
Comment (dustinxie) [inline action/protocol/execution/evm/evm.go:187]: i recall we now have value = 50? think we can remove this, no need to complicate things with this change.

**Review by millken — APPROVED** [review]

File: action/protocol/execution/evm/evm_test.go (hunk 1 above)
Comment (dustinxie) [reply action/protocol/execution/evm/evm_test.go:271]: i mean, `for` at L257

File: action/protocol/execution/evm/evm.go (hunk 2 above)
Comment (dustinxie) [reply action/protocol/execution/evm/evm.go:187]: `context.Random = &common.Hash{}` will do

**Review by dustinxie — APPROVED** [review]

**Pull request merged**